backend/src/main/java/de/laura/project/db/TempTriviaQuestionDB.java

```diff
@@ -17,38 +17,48 @@
 @NoArgsConstructor
 @Builder
 
-@Service
+@Repository
 public class TempTriviaQuestionDB {
 
-    private List<TriviaQuestionSet> triviaQuestionSetList;
+    private List<TriviaQuestionSet> triviaQuestionSetList = new ArrayList<>();
 
 
     public List<TriviaQuestionSet> returnQuestionList(List<TriviaQuestionSet> triviaQuestionSetList) {
-        this.triviaQuestionSetList = triviaQuestionSetList;
         return triviaQuestionSetList;
     }
 
-    public TriviaQuestionSetWithoutAnswer getSingleQuestion(int id) {
+    public TriviaQuestionSetWithoutCorrectAnswer getSingleQuestion(int questionId) {
 
-        List<TriviaQuestionSetWithoutAnswer> triviaQuestionSetListWithoutAnswer = new ArrayList<>();
+        List<TriviaQuestionSetWithoutCorrectAnswer> triviaQuestionSetListWithoutAnswer = new ArrayList<>();
 
         triviaQuestionSetListWithoutAnswer = triviaQuestionSetList.stream()
-                .map(triviaQuestionSet -> TriviaQuestionSetWithoutAnswer.builder()
+                .map(triviaQuestionSet -> TriviaQuestionSetWithoutCorrectAnswer.builder()
                         .id(triviaQuestionSet.getId())
                         .question(triviaQuestionSet.getQuestion())
                         .answers(triviaQuestionSet.getAnswers())
                         .build()).collect(Collectors.toList());
 
+        return triviaQuestionSetListWithoutAnswer.stream()
+                .filter(triviaQuestionSet -> triviaQuestionSet.getId() == (questionId))
+                .findFirst().get();
 
+        }
 
-        return
-
-        /*filter(triviaQuestionSet
-                -> triviaQuestionSet.getId() == (id))
-                .findFirst().get();
+    public boolean checkAnswer(int questionId, String selectedAnswer) {
```

Comment:
usually you would include any business logic within your service class and not DB class
your DB class should solely handle the connection to your DB getting, returning and setting new values within your DB 
anything containing logic should be moved to your service class